File: colt-steele/averagePair.js
// Write a function called averagePair. Given a sorted array of integers and a target average, determine if there is a pair of values in the array where the average of the pair equals the target average. There may be more than one pair that matches the average target.

//Attempt 1
function averagePair(arr, int){
  if (arr.length === 0) return false;
  let i = 0;
  let j = arr.length - 1;
  
  while (i < j) {
      let avg = (arr[i] + arr[j]) / 2
      if (avg > int) {
          j--
      } else if (avg < int) {
          i++
      } else if (avg === int) {
          return true;
      }
  }
  return false;
}

File: colt-steele/averagePair.rb
def average_pair(arr, int)
    if arr.length === 0 
        return false
    end
    i = 0
    j = arr.length - 1
    
    while i < j do
        avg = ((arr[i] + arr[j]) / 2)
        puts avg
        if avg > int 
            j -= 1
        avg = ((arr[i] + arr[j]) / 2)
        elsif avg < int
            i += 1
            avg = ((arr[i] + arr[j]) / 2)
        elsif avg === int
            return true
        end
    end
    false
end

puts average_pair([1,2,3], 2.5)